presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergAbstractMetadata.java

```diff
@@ -152,6 +162,8 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
 
     protected abstract ConnectorTableMetadata getTableMetadata(ConnectorSession session, SchemaTableName table);
 
+    protected abstract Table getIcebergTable(ConnectorSession session, SchemaTableName schemaTableName);
```

Comment:
I think if we are introducing  `getIcebergTable` then we can add this in the separate commit and make of use it even in the existing place where `getHiveIcebergTable` or `getNativeIcebergTable` is being called?

Reply:
Sure, I can remove this for now. I will open another PR for the refactor later.